### PR TITLE
PDF means

### PR DESCRIPTION
**Pull request by roystgnr**
This adds a centroid() method for sets and a distributionMean() method for PDFs.  Using the latter on a prior distribution ought to be useful for automatically choosing good initial iterates in inverse problems.

### CODE REVIEW

**Comment by dmcdougall** [comment]
I think this looks good.  Would you mind adding some tests for the means that don't throw exceptions?

**Comment by roystgnr** [comment]
"+226,-4"

**Comment by roystgnr** [comment]
"+1083,-4"

File: src/stats/src/ConcatenatedJointPdf.C [hunk 1]
```diff
@@ -183,6 +183,24 @@ ConcatenatedJointPdf<V,M>::lnValue(
...
+  std::vector<V*> vecs(m_densities.size(),(V*) NULL);
+  std::vector<double> values(m_densities.size(),0.);
+  double returnValue = 0.;
+  unsigned int cumulativeSize = 0;
+  for (unsigned int i = 0; i < vecs.size(); ++i) {
+    vecs[i] = new V(m_densities[i]->domainSet().vectorSpace().zeroVector());
+    m_densities[i]->distributionMean(*(vecs[i]));
+    meanVector.cwSet(cumulativeSize,*(vecs[i]));
+    cumulativeSize += vecs[i]->sizeLocal();
+    delete vecs[i];
```
Comment (dmcdougall) [inline src/stats/src/ConcatenatedJointPdf.C]: You might consider using a shared pointer for these.

File: src/basic/inc/BoxSubset.h [hunk 2]
```diff
@@ -66,6 +66,9 @@ class BoxSubset : public VectorSubset<V,M> {
    * m_minValues, and 2) all all components in \c vec are smaller than m_maxValues. */
   bool contains (const V& vec)     const;
 
+  //! Returns the centroid of this box subset.
```
Comment (dmcdougall) [inline src/basic/inc/BoxSubset.h]: Not sure how precise you want to be in these doxygen comments.  The method doesn't actually return anything, it fills up `vec`.

File: src/stats/src/ConcatenatedJointPdf.C (hunk 1 above)
Comment (roystgnr) [inline src/stats/src/ConcatenatedJointPdf.C]: Yeah, this is lousy code.  I was trying to follow the style in a couple of the other ConcatenatedJointPdf functions but I can just fix those instead.  :-)

Comment (dmcdougall) [inline src/stats/src/ConcatenatedJointPdf.C]: Don't worry about fixing up other parts of the file.  I can open a ticket for it so it's not forgotten.

**Pull request merged**